src/test/app/NFTokenBurn_test.cpp

```diff
@@ -786,22 +786,68 @@ class NFTokenBurn_test : public beast::unit_test::suite
         testBurnTooManyOffers(features);
     }
 
+protected:
+    void
+    run(std::uint32_t instance, bool last = false)
+    {
+        using namespace test::jtx;
+        static FeatureBitset const all{supported_amendments()};
+        static FeatureBitset const fixNFTDir{fixNFTokenDirV1};
+
+        static std::array<FeatureBitset, 4> const feats{
+            all - fixNonFungibleTokensV1_2 - fixNFTDir - fixNFTokenRemint,
+            all - fixNonFungibleTokensV1_2 - fixNFTokenRemint,
+            all - fixNFTokenRemint,
+            all};
+
+        if (BEAST_EXPECT(instance < feats.size()))
+        {
+            testWithFeats(feats[instance]);
+        }
+        BEAST_EXPECT(!last || instance == feats.size() - 1);
+    }
+
 public:
     void
     run() override
     {
-        using namespace test::jtx;
-        FeatureBitset const all{supported_amendments()};
-        FeatureBitset const fixNFTDir{fixNFTokenDirV1};
-
-        testWithFeats(
-            all - fixNonFungibleTokensV1_2 - fixNFTDir - fixNFTokenRemint);
-        testWithFeats(all - fixNonFungibleTokensV1_2 - fixNFTokenRemint);
-        testWithFeats(all - fixNFTokenRemint);
-        testWithFeats(all);
+        run(0);
+    }
+};
+
+class NFTokenBurn1_test : public NFTokenBurn0_test
```

Comment:
why is it designed to have subsequent test classes inheriting from `NFTokenBurn0_test`?

Why didn't we create a base class like `NFTokenBurnBase_Test`, which contains all the test cases and the new function `run(std::uint32_t instance, bool last = false)`.  And then have `NFTokenBurn0_test`, `NFTokenBurn1_test`, `NFTokenBurn2_test` and `NFTokenBurn3_test` all inherit from it?

Reply:
It could be done either way. The main reasons I chose to do it this way were
1. That's how I did it in `TxQ_test.cpp`
2. So that the base test class would be useful as a test in and of itself.

If you have a strong opinion, I can change it, but I don't think it really matters.